packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -74,51 +74,54 @@ export function useCurrentMergeThemeStyleVariationsWithUserConfig( {
 		const clonedUserVariation = cloneDeep( userVariation );
 
 		// Get user variation and remove the settings for the given property.
-		const userVariationWithoutProperty = removePropertyFromObject(
+		const userVariationWithoutProperties = removePropertiesFromObject(
 			clonedUserVariation,
-			property
+			properties
 		);
-		userVariationWithoutProperty.title = __( 'Default' );
+		userVariationWithoutProperties.title = __( 'Default' );
 
-		const variationsWithSinglePropertyAndBase = variationsFromTheme
+		const variationsWithPropertiesAndBase = variationsFromTheme
 			.filter( ( variation ) => {
-				return isVariationWithSingleProperty( variation, property );
+				return isVariationWithProperties( variation, properties );
 			} )
 			.map( ( variation ) => {
 				return mergeBaseAndUserConfigs(
-					userVariationWithoutProperty,
+					userVariationWithoutProperties,
 					variation
 				);
 			} );
 
 		return [
-			userVariationWithoutProperty,
-			...variationsWithSinglePropertyAndBase,
+			userVariationWithoutProperties,
+			...variationsWithPropertiesAndBase,
 		];
-	}, [ property, userVariation, variationsFromTheme ] );
+	}, [ properties, userVariation, variationsFromTheme ] );
```

Comment:
I'm not familiar with this code at all but there's something that is probably not right here. If you're using properties as a dependency for a useMemo call, I'm guessing that you expect the same object instance to be returned here if the properties don't change (maybe for some performance reasons), but here it's almost never the case since properties is passed as an inline array, so properties will always trigger a recompute and a new instance returned by useMemo.

Reply:
Ah yes good point. Would it work to spread the properties in the dependency array, like this?

```suggestion
	}, [ ...properties, userVariation, variationsFromTheme ] );
```

Reply:
That wouldn't work either, because the size of the properties array might change and I believe it's not allowed in these react hooks to have a variable size.

Reply:
What if we convert the array to a string?

Reply:
It would work I think.

Reply:
Done in [ea0be03](https://github.com/WordPress/gutenberg/pull/62847/commits/ea0be0360b6e1372d22a1cf84aba58800be12e39)